app/code/Magento/Backend/Block/Widget/Grid/Column/Renderer/AbstractRenderer.php

```diff
@@ -139,9 +139,10 @@ public function renderHeader()
     {
         if (false !== $this->getColumn()->getSortable()) {
             $className = 'not-sort';
-            $dir = strtolower($this->getColumn()->getDir());
+            $dir = $this->getColumn()->getDir() ? strtolower($this->getColumn()->getDir()) : '';
```

Comment:
```suggestion
            $dir = strtolower($this->getColumn()->getDir() ?: '');
```

Reply:
Hello @ihor-sviziev 
I think this is redundant operation `strtolower('')`;
I think @anzin added correct fix here.

Reply:
As discussed in Slack - some of these parts are called more than 100k times, so it's better to keep a more optimized version there.

I agree with that.

Reply:
Let's remove the cases when it might be not a string (int, float, object, null, etc)
```suggestion
            $dir = is_string($this->getColumn()->getDir()) ? strtolower($this->getColumn()->getDir()) : '';
```